builder/bundleset.go

```diff
@@ -145,22 +147,17 @@ func validateBundleFile(filename string, lvl ValidationLevel) error {
 		return errors.New(errText)
 	}
 
-	if lvl == BasicValidation {
-		if errText != "" {
-			return errors.New(strings.TrimSuffix(errText, "\n"))
-		}
-		return nil
-	}
-
 	// Strict Validation
-	err = validateBundle(b)
-	if err != nil {
-		errText += err.Error() + "\n"
-	}
+	if lvl == StrictValidation {
```

Comment:
All this logic should be replaced with a
```go
switch {
case lvl >= basicValidation:
    // basic validation only
    fallthrough
case lvl >= validateName:
    // only validate the name of the bundle
    fallthrough
case lvl >= validateSomeOtherSpecificThing:
    // only validate some other specific thing
    fallthrough
case lvl >= strictValidation:
    // additional strict validation
default:
    // no validation?
}
```
Now this means you have actual *levels* instead of being able to select validation. If you want to let the user select validation you can essentially use the same switch just replace the `>=` with `==`.

Reply:
Wouldn't `fallthrough` execute the first statement in the next `case`? E.g if lvl is `BasicValidation`, all the cases including `StrictValidation` will be executed.

Reply:
eh maybe I have the cases backwards oops. Yeah `StrictValidation` would be first and would `fallthrough` to the basic validation.